ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientOptionsImpl.java

```diff
@@ -158,6 +164,12 @@ public ClientOptions setHttp2MultiplexingLimit(final int http2MultiplexingLimit)
     return this;
   }
 
+  @Override
+  public ClientOptions setRequestHeaders(final Map<String, String> requestHeaders) {
+    this.requestHeaders = ImmutableMap.copyOf(requestHeaders);
```

Comment:
Do we want to catch for null input or is the NPE that'd come out of `copyOf` what we want?

Reply:
Yup, fixed.